ledger/src/shred.rs

```diff
@@ -467,8 +473,36 @@ impl Shred {
         self.common_header.signature
     }
 
-    pub fn seed(&self) -> [u8; 32] {
+    pub fn seed(
+        &self,
+        leader_schedule_cache: &Arc<LeaderScheduleCache>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> [u8; 32] {
         let mut seed = [0; 32];
+
+        let bf = bank_forks.read().unwrap();
+        let maybe_bank = bf.get(self.slot());
+        if let Some(bank) = maybe_bank {
+            if enable_deterministic_seed(self.slot(), bank) {
+                let maybe_leader_pubkey =
+                    leader_schedule_cache.slot_leader_at(self.slot(), Some(bank));
+                let h = match maybe_leader_pubkey {
+                    Some(pubkey) => hashv(&[
+                        &self.slot().to_le_bytes(),
+                        &self.index().to_le_bytes(),
+                        &pubkey.to_bytes(),
```

Comment:
probably this should be `as_bytes` instead.

Reply:
Is this referring to the return value of hashv?

Reply:
I was referring to `pubkey.to_bytes()` but, nevermind, it seems like `Pubkey` does not implement `as_bytes`